tests/components/mqtt/test_common.py

```diff
@@ -1690,3 +1690,56 @@ async def help_test_reloadable_late(hass, caplog, tmp_path, domain, config):
     assert hass.states.get(f"{domain}.test_new_1")
     assert hass.states.get(f"{domain}.test_new_2")
     assert hass.states.get(f"{domain}.test_new_3")
+
+
+async def help_test_setup_manual_entity_from_yaml(
+    hass,
+    caplog,
+    tmp_path,
+    platform,
+    config,
+    assert_entity_exists=True,
+    remove_platform=True,
+):
+    """Test setup from yaml through configuration entry.
+
+    This helper has some optional parameters.
+    With the default settings a manual MQTT item will be setup with
+    the new config schema under the `mqtt->platform->[manual_config]` key.
+    A `platform` key will be removed from the config if the new schema is used
+    and `remove_platform` is set, this way we can share the PLATFORM_SCHEMA.
+    - `assert_entity_exists` adds a check to assert an entity with name platform.test exists.
+    - `remove_platform` mocks the forwarding og the entry setup to the platforms.
+    - `legacy_schema` uses the deprecated config schema to setup the manual MQTT item.
+    """
+    configs = []
+    if config:
+        yaml_config = copy.deepcopy(config)
+        if assert_entity_exists:
+            yaml_config["name"] = "test"
+        if remove_platform:
+            del yaml_config["platform"]
+        configs.append(yaml_config)
+    config_structure = {mqtt.DOMAIN: {platform: configs}}
+    # Remove platform key from the config since this is not valid here
+    new_yaml_config_file = tmp_path / "configuration.yaml"
+    new_yaml_config = yaml.dump(config_structure)
+    new_yaml_config_file.write_text(new_yaml_config)
+    assert new_yaml_config_file.read_text() == new_yaml_config
+
+    await async_setup_component(hass, mqtt.DOMAIN, config_structure)
+    # Mock config entry
+    entry = MockConfigEntry(domain=mqtt.DOMAIN, data={mqtt.CONF_BROKER: "test-broker"})
+    entry.add_to_hass(hass)
+
+    with patch.object(hass_config, "YAML_CONFIG_FILE", new_yaml_config_file), patch(
```

Comment:
Why are we patching the config file?

Reply:
I agree it should not be needed. Instead of this, we should setup the mqtt component with the wanted config before setting up the mqtt config entry.

Reply:
The configuration is read from configuration.yaml. To be able to test a config we need to patch it.

Reply:
Where is it read from configuration.yaml? We manually setup the mqtt component and config entry?

Reply:
It is not (anymore) I'll update the helper.